mesonbuild/dependencies.py

```diff
@@ -1568,6 +1568,118 @@ def __init__(self, environment, kwargs):
+        'llvm-config-3.4', 'llvm-config34', 'llvm-config-3.3', 'llvm-config33',
+        'llvm-config-devel',
+    ]
+    llvmconfig = None
+    _llvmconfig_found = False
+    __best_found = None
+
+    # TODO: should we handle components like clang?
+    def __init__(self, environment, kwargs):
+        super().__init__('llvm-config', kwargs)
+        self.cargs = []
+        self.libs = []
+        self.components = []
+
+        req_version = kwargs.get('version', None)
+        if self.llvmconfig is None:
+            self.check_llvmconfig(req_version)
+        if not self._llvmconfig_found:
+            if self.__best_found is not None:
+                mlog.log('Dependency LLVM found:', mlog.red('NO'))
+                mlog.log('found {!r} but need:'.format(self.modversion),
+                         req_version)
+            else:
+                mlog.log("No llvm-config found; can't detect dependency")
+            return
+
+        p, out = Popen_safe([self.llvmconfig, '--version'])[:2]
+        if p.returncode != 0:
+            mlog.log('Dependency LLVM found:', mlog.red('NO'))
+        else:
+            self.modversion = out.strip()
+            mlog.log('Dependency LLVM found:', mlog.green('YES'))
+            self.is_found = True
+
+            p, out = Popen_safe([self.llvmconfig, '--libs'])[:2]
+            if p.returncode != 0:
+                raise DependencyException('Could not generate libs for LLVM.')
+            self.libs = out.split()
+
+            p, out = Popen_safe([self.llvmconfig, '--includedir'])[:2]
+            if p.returncode != 0:
+                raise DependencyException('Could not generate includedir for LLVM.')
+            self.include_directories = out.split()
+
+            p, out = Popen_safe([self.llvmconfig, '--components'])[:2]
+            if p.returncode != 0:
+                raise DependencyException('Could not generate components for LLVM.')
+            self.components = out.split()
+
+        components = kwargs.get('components')
+        if components:
+            self.validate_components(components)
+
+    def validate_components(self, components):
+        if components:
+            if isinstance(components, str):
+                if components not in self.components:
+                    mlog.log('LLVM component', components, 'found:', mlog.red('NO'))
+                    self.is_found = False
+                else:
+                    mlog.log('LLVM component', components, 'found:', mlog.green('YES'))
+
+            elif isinstance(components, list):
+                for comp in components:
+                    self.validate_components(comp)
+            else:
+                raise DependencyException(
+                    'LLVM components must be either strings or lists of strings')
+
+    def get_version(self):
+        return self.modversion
+
+    def get_compile_args(self):
+        return self.cargs
```

Comment:
You never fetch this from llvm-config, so this will always return an empty array. And I can see why because `--cflags` and `--cxxflags` are full of garbage. But `--cppflags` seems to return useful things, so maybe use that? It's also a superset of `--includedir`, so you can then get rid of `self.include_directories`.